quickwit/quickwit-codegen/src/codegen.rs

```diff
@@ -89,6 +89,85 @@ impl Codegen {
         }
         Ok(())
     }
+
+    pub fn builder() -> CodegenBuilder {
+        CodegenBuilder::default()
+    }
+}
+
+#[derive(Default)]
+pub struct CodegenBuilder {
+    protos: Vec<String>,
+    includes: Vec<String>,
+    out_dir: String,
+    prost_config: ProstConfig,
+    result_type_path: String,
+    error_type_path: String,
+    generate_extra_service_methods: bool,
+    generate_prom_label_for_requests: bool,
+}
+
+impl CodegenBuilder {
+    pub fn with_protos(mut self, protos: &[&str]) -> Self {
+        self.protos = protos.iter().map(|p| p.to_string()).collect();
```

Comment:
nit: we like long variable names :)

```suggestion
        self.protos = protos.iter().map(|proto| proto.to_string()).collect();
```

Reply:
Then long variable names you shall have :)